python/python_fastapi_server/routers/ogcapi_tools.py

```diff
@@ -0,0 +1,446 @@
+        lon = bbox[0] + lonval * dlon + dlon / 2.0
+        for latval in range(nlat):
+            lat = bbox[1] + latval * dlat + dlat / 2
+            coords.append([lon, lat])
+    return coords
+
+
+def call_adaguc(url):
+    """Call adaguc-server"""
+    adaguc_instance = setup_adaguc()
+
+    url = url.decode()
+    if "?" in url:
+        url = url[url.index("?") + 1 :]
+    logger.info(url)
+    stage1 = time.perf_counter()
+
+    adagucenv = {}
+
+    # Set required environment variables
+    adagucenv["ADAGUC_ONLINERESOURCE"] = (
+        os.getenv("EXTERNALADDRESS", "http://192.168.178.113:8080") + "/adaguc-server?"
+    )
+    adagucenv["ADAGUC_DB"] = os.getenv(
+        "ADAGUC_DB", "user=adaguc password=adaguc host=localhost dbname=adaguc"
+    )
+
+    # Run adaguc-server
+    # pylint: disable=unused-variable
+    status, data, headers = adaguc_instance.runADAGUCServer(
+        url, env=adagucenv, showLogOnError=True
+    )
+
+    # Obtain logfile
+    logfile = adaguc_instance.getLogFile()
+    adaguc_instance.removeLogFile()
+
+    stage2 = time.perf_counter()
+    logger.info("[PERF] Adaguc execution took: %f", (stage2 - stage1))
+
+    if len(logfile) > 0:
+        logger.info(logfile)
+
+    return status, data
+
+
+# @cacher.memoize(timeout=30)
+def get_capabilities(collname):
+    """
+    Get the collectioninfo from the WMS GetCapabilities
+    """
+    coll = generate_collections().get(collname)
+    if "dataset" in coll:
+        logger.info("callADAGUC by dataset")
+        dataset = coll["dataset"]
+        urlrequest = (
+            f"dataset={dataset}&service=wms&version=1.3.0&request=getcapabilities"
+        )
+        status, response = call_adaguc(url=urlrequest.encode("UTF-8"))
+        if status == 0:
+            xml = response.getvalue()
+            wms = WebMapService(coll["service"], xml=xml, version="1.3.0")
+        else:
+            logger.error("status: %d", status)
+            return {}
+    else:
+        logger.info("callADAGUC by service %s", coll)
+        wms = WebMapService(coll["service"], version="1.3.0")
+    return wms.contents
+
+
+# @cacher.cached(timeout=30, key_prefix="collections")
+def generate_collections():
+    """
+    Generate OGC API Feature collections
+    """
+    collections = get_datasets(os.environ.get("ADAGUC_DATASET_DIR"))
+    return collections
+
+
+def get_dimensions(layer, skip_dims=None):
+    """
+    get_dimensions
+    """
+    dims = []
+    for dim_name in layer.dimensions:
+        if not dim_name in skip_dims:
+            dim_name = {
+                "name": dim_name,
+                "values": layer.dimensions[dim_name]["values"],
+            }
+            dims.append(dim_name)
+    return dims
+
+
+# @cacher.memoize(timeout=30)
+def get_parameters(collname):
+    """
+    get_parameters
+    """
+    contents = get_capabilities(collname)
+    layers = []
+    for layer in contents:
+        dims = get_dimensions(contents[layer], ["time"])
```

Comment:
This gets all dimensions except for the time dimension?

Reply:
Yes